src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -206,14 +241,21 @@ public override bool Equals(object obj)
             CultureAwareComparer comparer = obj as CultureAwareComparer;
             return
                 comparer != null &&
-                _ignoreCase == comparer._ignoreCase &&
+                _options == comparer._options &&
                 _compareInfo.Equals(comparer._compareInfo);
         }
 
         public override int GetHashCode()
         {
             int hashCode = _compareInfo.GetHashCode();
-            return _ignoreCase ? ~hashCode : hashCode;
+            return (_options & CompareOptions.IgnoreCase) != 0 ? ~hashCode : hashCode;
```

Comment:
hashCode should include all `_options`, not just ignore case.

Reply:
```
  return _options.GetHashCode() | _compareInfo.GetHashCode();
```
is this okay ?